src/devices/sound/k051649.cpp

```diff
@@ -115,11 +115,18 @@ void k051649_device::device_clock_changed()
 	uint32_t old_rate = m_rate;
 	m_rate = clock()/16;
 	m_mclock = clock();
-
-	// allocate a buffer to mix into - 1 second's worth should be more than enough
+	
+	if (old_rate < m_rate)
+	{
+		memset(&m_mixer_buffer[0], 0, sizeof(m_mixer_buffer));
```

Comment:
Same issue as on the other pull request - `sizeof` has nothing to do with the vector's content.